test/new-e2e/utils/e2e/e2e.go

```diff
@@ -91,38 +97,69 @@ func NewSuite[Env any](stackName string, stackDef *StackDefinition[Env], options
 	return &testSuite
 }
 
+func DevMode[Env any]() func(*Suite[Env]) {
+	return func(suite *Suite[Env]) {
+		suite.devMode = true
+	}
+}
+
+func SkipDeleteOnFailure[Env any]() func(*Suite[Env]) {
+	return func(suite *Suite[Env]) {
+		suite.skipDeleteOnFailure = true
+	}
+}
+
+// Env returns the current environment.
+// In order to improve the efficiency, this function behaves as follow:
+//   - It creates the default environment if no environment exists. It happens only during the first call of the test suite.
+//   - It restores the default environment if UpdateEnv was not already be called during this test.
+//     This avoid having to restore the default environment for each test even if UpdateEnv immedialy
+//     overrides this environment.
+func (suite *Suite[Env]) Env() *Env {
```

Comment:
I think we should keep `Env()` otherwise accessing a VM would be `suite.GetCurrentEnv().VM` instead of `suite.Env().VM`. What do you think?

Reply:
What I don't like about `Env` is that it looks like a getter, but it can potentially change the env. The name is hiding this, but I can see that having the comment will bring more context. Not a blocker !

Reply:
I tend to not use `GetSomething` when the function is more than a simple getter. A better name could be `RestoreEnvIfNotChanged()`. From a user perspective, this is a detail of implementation as the function `Env()` behaves as if it was a simple getter and the default env is restored before running each test. Does it make sense?